### PR TITLE
Support per-user checkins

### LINKED ISSUE #3: Create some endpoints to query a user's active scheduled checkins

### PR DESCRIPTION
**Pull request by mattddean**
This pull request introduces the concept of supporting multiple users.

- Update rule naming scheme: `trigger-checkin-${userId}-${crc32}`
  - Including the user id allows us to store checkins per-user and then look up the checkins for that user
  - It's the responsibility of the frontend system that interacts with this service to provide a user id for each user, but checkin-service is now capable of responding with a user's scheduled checkins given a user id.
- Add an endpoint to list the checkins for a user id.

Resolves #3

### CODE REVIEW

**Comment by mattddean** [comment]
@aaron-pham would value your feedback!

File: src/handlers/handle-list-scheduled-checkins-for-user.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+type RequestQueryParams = {
+  user_id: string;
+};
+
+interface ResponseBody {
+  data: Checkin[];
+}
+
+/**
+ * Check the user into a single flight.
```
Comment (aaron-pham) [inline src/handlers/handle-list-scheduled-checkins-for-user.ts:20]: This comment should be different right?

File: src/handlers/handle-scheduled-checkin.ts [hunk 2]
```diff
@@ -92,6 +92,8 @@ async function handleInternal(event: AWSLambda.SQSEvent) {
     Luxon.DateTime.now().toLocaleString(Luxon.DateTime.DATETIME_FULL_WITH_SECONDS)
   );
 
+  // Perform the checkin. Since we've already gotten a response indicating that the checkin is
+  // ready, we only need to make one request.
```
Comment (aaron-pham) [inline src/handlers/handle-scheduled-checkin.ts:96]: Nice comments, I've always mentally visualized that the above step is actually doing the check-in, I didn't realize it's just checking if it's ready and then we do the check-in after the fact.

**Review by aaron-pham — APPROVED** [review]

**Comment by aaron-pham** [comment]
Looks great! So is userId a required field now for things to work? 

Is the idea that the frontend will generate the userId based on the auth it's using?

**Comment by mattddean** [comment]
> Looks great! So is userId a required field now for things to work? 
> 
> 
> 
> Is the idea that the frontend will generate the userId based on the auth it's using? 

Yeah, exactly. But frontends that aren't ready to support multiple users could simply provide a hardcoded User ID that's the same for everyone who visits the site.

**Pull request merged**